eth-tests/src/main/java/tech/pegasys/teku/ethtests/finder/ShufflingTestFinder.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.ethtests.finder;
+
+import com.google.errorprone.annotations.MustBeClosed;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+public class ShufflingTestFinder implements TestFinder {
```

Comment:
Would it make sense for these `TestFinder` classes to throw if they don't find any tests? Since that is unexpected.

Reply:
It's expected that they won't find tests in some cases - for example there are no shuffling tests under `general` and no genesis tests under `mainnet`.